### PR TITLE
PLT-299:Updated is_system_node_pool field to optional

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: examples/resources/spectrocloud_cluster_azure/resource.tf [hunk 1]
```diff
@@ -35,7 +34,6 @@ resource "spectrocloud_cluster_azure" "cluster" {
   }
 
   machine_pool {
-    is_system_node_pool = true
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_azure/resource.tf:38]: please keep it in the example.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: examples/resources/spectrocloud_cluster_azure/resource.tf (hunk 1 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_azure/resource.tf:38]: okay

Comment (nikchern) [reply examples/resources/spectrocloud_cluster_azure/resource.tf:38]: @SivaanandM let's also set a default value in golang. I assume it will be false.

Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_azure/resource.tf]: updated

File: spectrocloud/resource_cluster_azure.go [hunk 2]
```diff
@@ -205,7 +205,8 @@ func resourceClusterAzure() *schema.Resource {
 						},
 						"is_system_node_pool": {
 							Type:        schema.TypeBool,
-							Required:    true,
+							Optional:    true,
+							Default:     false,
 							Description: "Whether this machine pool is a system node pool.",
```
Comment (nikchern) [inline spectrocloud/resource_cluster_azure.go:210]:
```suggestion
							Description: "Whether this machine pool is a system node pool. Default value is `false'.",
```

**Review by nikchern — CHANGES_REQUESTED** [review]

**Review by nikchern — APPROVED** [review]

**Pull request merged**